app/src/main/java/pm/gnosis/heimdall/ui/onboarding/password/PasswordSetupActivity.kt

```diff
@@ -60,7 +62,7 @@ class PasswordSetupActivity : SecuredBaseActivity() {
 
     private fun onPasswordValidation(passwordValidation: PasswordValidation) {
         when (passwordValidation) {
-            is PasswordValid -> startActivity(PasswordConfirmActivity.createIntent(this, passwordValidation.password))
+            is PasswordValid -> startActivity(PasswordConfirmActivity.createIntent(this, Sha3Utils.keccak(passwordValidation.password.toByteArray())))
```

Comment:
Why not set the hash in the PasswordValidation so that it can be calculated in the ViewModel